lib/Mojolicious/Guides/Tutorial.pod

```diff
@@ -0,0 +1,938 @@
+  };
+
+  app->start;
+  __DATA__
+
+  @@ form.html.ep
+  <!DOCTYPE html>
+  <html>
+    <head><title>Upload</title></head>
+    <body>
+      %= form_for upload => (enctype => 'multipart/form-data') => begin
+        %= file_field 'example'
+        %= submit_button 'Upload'
+      % end
+    </body>
+  </html>
+
+To protect you from excessively large files there is also a limit of 16MiB by default, which you can tweak with the
+attribute L<Mojolicious/"max_request_size">.
+
+  # Increase limit to 1GiB
+  app->max_request_size(1073741824);
+
+=head2 User agent
+
+With L<Mojo::UserAgent>, which is available through the helper L<Mojolicious::Plugin::DefaultHelpers/"ua">, there's a
+full featured HTTP and WebSocket user agent built right in. Especially in combination with L<Mojo::JSON> and
+L<Mojo::DOM> this can be a very powerful tool.
+
+  use Mojolicious::Lite -signatures;
+
+  # Blocking
+  get '/headers' => sub ($c) {
+    my $url = $c->param('url') || 'https://mojolicious.org';
+    my $dom = $c->ua->get($url)->result->dom;
+    $c->render(json => $dom->find('h1, h2, h3')->map('text')->to_array);
+  };
+
+  # Non-blocking
+  get '/title' => sub ($c) {
+    $c->ua->get('mojolicious.org' => sub ($ua, $tx) {
+      $c->render(data => $tx->result->dom->at('title')->text);
+    });
+  };
+
+  # Concurrent non-blocking
+  get '/titles' => sub ($c) {
+    my $mojo = $c->ua->get_p('https://mojolicious.org');
+    my $cpan = $c->ua->get_p('https://metacpan.org');
+    Mojo::Promise->all($mojo, $cpan)->then(sub ($mojo, $cpan) {
+      $c->render(json => {
+        mojo => $mojo->[0]->result->dom->at('title')->text,
+        cpan => $cpan->[0]->result->dom->at('title')->text
+      });
+    })->wait;
+  };
+
+  app->start;
+
+For more information about the user agent see also L<Mojolicious::Guides::Cookbook/"USER AGENT">.
+
+=head2 WebSockets
+
+WebSocket applications have never been this simple before. Just receive messages by subscribing to events such as
+L<Mojo::Transaction::WebSocket/"json"> with L<Mojolicious::Controller/"on"> and return them with
+L<Mojolicious::Controller/"send">.
+
+  use Mojolicious::Lite -signatures;
+
+  websocket '/echo' => sub ($c) {
+    $c->on(json => sub ($c, $hash) {
+      $hash->{msg} = "echo: $hash->{msg}";
+      $c->send({json => $hash});
+    });
+  };
+
+  get '/' => 'index';
+
+  app->start;
+  __DATA__
+
+  @@ index.html.ep
+  <!DOCTYPE html>
+  <html>
+    <head>
+      <title>Echo</title>
+      <script>
+        const ws = new WebSocket('<%= url_for('echo')->to_abs %>');
+        ws.onmessage = function (event) {
+          document.body.innerHTML += JSON.parse(event.data).msg;
+        };
+        ws.onopen = function (event) {
+          ws.send(JSON.stringify({msg: 'I ♥ Mojolicious!'}));
+        };
+      </script>
+    </head>
+  </html>
+
+For more information about real-time web features see also L<Mojolicious::Guides::Cookbook/"REAL-TIME WEB">.
+
+=head2 Mode
```

Comment:
There's a lot of good Mojo::Log info here, but looking at the TOC one would not think to come to this section for that Mojo::Log info being looked for.  Rename this to "Mode and Log" or something?

Related but _way_ off topic...  I just watched this really neat [SQLite video](https://www.youtube.com/watch?v=w0FTWcuyliA) and the topic was using vectors and OpenAI to automatically create related page links.  The search bar in mojolicious.org which uses google.com is not really that great IMO.  Google just isn't as good as they once were...!  It might be a really great feature (and example!) to add an OpenAI-related technique to mojolicious.org and provide some really helpful related topic links!  That alone could address a lot of the comments I have on this PR!